custom_model_runner/datarobot_drum/drum/model_adapter.py

```diff
@@ -79,6 +85,10 @@ def __init__(self, model_dir, target_type=None):
         self._custom_task_class = None
         self._custom_task_class_instance = None
 
+    def _log_and_raise_final_error(self, exc: Exception, message: str) -> NoReturn:
+        self._logger.error(f"{message} Exception: {exc!r}")
+        raise DrumPythonModelAdapterError(f"{message} Exception: {exc!r}")
```

Comment:
will the traceback of the original exception be printed to logs?

e.g.
- don't we need `exc_info=True` for logger?
- don't we need `raise DrumPythonModelAdapterError(...) from exc` here?

Reply:
you want `logger.exception`

https://github.com/python/cpython/blob/3.12/Lib/logging/__init__.py#L1558

```python
    def error(self, msg, *args, **kwargs):
        """
        Log 'msg % args' with severity 'ERROR'.

        To pass exception information, use the keyword argument exc_info with
        a true value, e.g.

        logger.error("Houston, we have a %s", "major problem", exc_info=1)
        """
        if self.isEnabledFor(ERROR):
            self._log(ERROR, msg, args, **kwargs)

    def exception(self, msg, *args, exc_info=True, **kwargs):
        """
        Convenience method for logging an ERROR with exception information.
        """
        self.error(msg, *args, exc_info=exc_info, **kwargs)
```

Reply:
you shouldn't need the `from` to get the stacktrace, but it shouldn't hurt either.